### PR TITLE
Add acorn-loose parser

### PR DESCRIPTION
**Pull request by TimothyGu**

### CODE REVIEW

File: src/parsers/js/acorn-loose.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+  displayName: ID,
+  version: `${pkg.version}`,
+  homepage: pkg.homepage,
+  locationProps: new Set(['range', 'loc', 'start', 'end']),
+
+  loadParser(callback) {
+    let _localReq = require;
+    require(['acorn'], (acorn) => {
+      global.acorn = acorn;
+      _localReq(['acorn/dist/acorn_loose'], callback);
```
Comment (TimothyGu) [inline src/parsers/js/acorn-loose.js]: The `acorn/dist/acorn_loose.js` file is built with Browserify, which means that the Webpack `require` function cannot get in there. Hence these hacks.

**Comment by fkling** [comment]
I feel like this should rather be an option in the acorn parser settings, and based on that option it would choose which parser to use. Other than avoiding to load the loose parser if it's not needed, I don't see an advantage to have it as a separate entry.

Thoughts?

**Comment by hzoo** [comment]
Yeah, I think that would be good even though its a separate file/etc. https://github.com/ternjs/acorn#distacorn_loosejs

**Comment by TimothyGu** [comment]
@fkling @hzoo Done.

File: src/parsers/js/acorn.js [hunk 2]
```diff
@@ -24,16 +24,33 @@ export default {
   locationProps: new Set(['range', 'loc', 'start', 'end']),
 
   loadParser(callback) {
+    const parsers = {};
+    const _localReq = require;
+
     require(['acorn', 'acorn-jsx/inject'], (acorn, jsxInject) => {
```
Comment (fkling) [inline src/parsers/js/acorn.js]: I think we should be able to directly require `src/loose/index.js` here and avoid all the issues with loading the built version?

Comment (fkling) [inline src/parsers/js/acorn.js]: Also, we should probably do the same thing with acorn itself, but that doesn't have to be done in this PR.

Comment (TimothyGu) [inline src/parsers/js/acorn.js]: Fixed.

**Pull request merged**